data_structures/binary_tree/segment_tree.py

```diff
@@ -3,29 +3,57 @@
 
 class SegmentTree:
     def __init__(self, a):
-        self.N = len(a)
+        self.A = a
+        self.N = len(self.A)
         self.st = [0] * (
             4 * self.N
         )  # approximate the overall size of segment tree with array N
```

Comment:
# approximate the overall size of the segment tree with array N
```suggestion
        self.st = [0] * (4 * self.N)
```

Reply:
Agreed, I'll make an issue for people to tackle naming issues in the repository, seems like it was an older thing